### PR TITLE
draft - Search Functionality

### LINKED ISSUE #5: search functionality
user able to search for ingredients/meals and populate on page

### PR DESCRIPTION
**Pull request by kmg37**
Created beginnings of a search bar that's currently using JSONPlaceholder to test API requests.

Components added:

- SearchBar
- SearchResult
- SearchResultsList
- SearchFunction

As well as their corresponding css files.

### CODE REVIEW

**Comment by kmg37** [comment]
Not looking to merge yet

**Comment by royemosby** [comment]
@kmg37 One way that I have seen WIP PRs be handled is to prefix the title with "draft". That will let a reviewer know, at a glance, that the PR is not to be merged.

Also, could you include a summary of changes in the PR request? It helps me understand what changes you are making with the code you are introducing.

**Comment by kmg37** [comment]
@royemosby Fair points! Good looking out on that, thank you.

File: package.json [hunk 1]
```diff
@@ -11,8 +11,10 @@
   },
   "dependencies": {
     "axios": "^1.6.7",
+    "front-end-template": "file:",
```
Comment (royemosby) [inline package.json:14]: what does this do?

Comment (kmg37) [reply package.json:14]: Ooh nothing to my knowledge. I had issues with getting the project running at the beginning, was trying a few different things to get it going, may have been from that.

Comment (royemosby) [reply package.json:14]: Gotcha - if you could, remove when you're ready for the PR to be reviewed.

Comment (kmg37) [reply package.json:14]: Will do, thanks for the heads up.

**Review by royemosby — COMMENTED** [review]
There are too many conflicts for me to resolve - especially with files that should not have been changed. I am going to close this PR out as-is and leave the `5-search-functionality` branch intact so we can manually recover code as needed.

**Pull request closed without merging**